src/uwtools/set_config.py

```diff
@@ -94,7 +117,13 @@ def create_config_obj(argv):
         config_obj.update_values(user_config_obj)
 
     if user_args.outfile:
-        config_obj.dump_file(user_args.outfile)
+        outfile_type = user_args.out_file_type or pathlib.Path(user_args.outfile).suffix
```

Comment:
For consistency, maybe this one should also call `get_file_type`?

Reply:
There is a weird issue on outfile, as if the outfile is set to a temp directory it will fail os.path.exists(arg) - let me see if I can sort it under get_file_type

Reply:
Got it - get_file_type doesn't need to call path_if_file_exists as that is already checked by the arg call